core/src/main/java/com/yahoo/oak/Block.java

```diff
@@ -78,8 +78,8 @@ void clean() {
         cleaner.clean();
     }
 
-    void readByteBuffer(Slice s) {
-        s.setBuffer(buffer);
+    void setSliceBuffer(Slice s) {
+        s.setBuffer(buffer, id);
```

Comment:
I don't understand the justification for this modification.

Reply:
it is in order to check the new block id with the old